### PR TITLE
Vet clinic database: add join table for visits

### PR DESCRIPTION
**Pull request by melkamu12**
### Vet Clinic Project:

In this PR, I added I create table Animals.

 ## **Activities done**
###  **Database Schema**:
 * [ ] Created a table named vets with columns id, name, age, and date_of_graduation.
 * [ ] Created a join table named specializations to handle the many-to-many relationship between vets and species.
 * [ ] Created a join table named visits to handle the many-to-many relationship between animals and vets, including the date of the visit.
### **Data Insertion:**
 * [ ] Inserted data for vets William Tatcher, Maisy Smith, Stephanie Mendez, and Jack Harkness, along with their graduation details.
 * [ ] Inserted data for vet specializations as provided in the project requirements.
 * [ ] Inserted data for animal visits to various vets, including the date of the visit.
 
 ## **screenshot** 

 ## **Checklist**:
 * [ ] Follows the style guidelines of this project
 * [ ] follow gitflow

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -61,5 +61,24 @@ SELECT a.name FROM animals a JOIN species s ON a.species_id = s.id JOIN owners o
...
+--Details for most recent visit: animal information, vet information, and date of visit.
+SELECT a.name, vs.date_of_visit, v.name FROM visits vs JOIN animals a ON a.id = vs.animal_id JOIN vets v ON v.id = vs.vet_id ORDER BY vs.date_of_visit DESC LIMIT 1;
+--How many visits were with a vet that did not specialize in that animal's species?
+SELECT COUNT(*) AS not_specialty FROM visits v JOIN vets vet ON v.vet_id = vet.id JOIN animals a ON v.animal_id = a.id LEFT JOIN specializations sp ON vet.id = sp.vets_id AND a.species_id = sp.species_id WHERE sp.vets_id IS NULL;
+--What specialty should Maisy Smith consider getting? Look for the species she gets the most.
+SELECT s.name AS suggested_specialty, COUNT(*) AS animal_count FROM animals a JOIN visits v ON a.id = v.animal_id JOIN vets vt ON v.vet_id = vt.id JOIN vets ms ON ms.name = 'Maisy Smith' JOIN specializations sp ON vt.id = sp.vets_id AND a.species_id = sp.species_id JOIN species s ON sp.species_id = s.id WHERE ms.id = vt.id GROUP BY s.name ORDER BY animal_count DESC LIMIT 1;
```
Comment (franclobo) [inline queries.sql]: - [ ] The result in the picture doesn't show the correct answer. What specialty should Maisy Smith consider getting? Look for the species she gets the most. I suggest grouping the animal visits by species, there are nine visits to Maisy Smith, three Pokemon, and six digimon.

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @melkamu12, 

# Required Changes ♻️

so-close-jim-halpert

Good job so far!
You are doing it well 👏🏽 there are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

- Correct schema ✔️ 
- Correct data ✔️ 
- Correct queries ✅ 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by melkamu12** [comment]
image
Fixing with this

**Review by sumairq — APPROVED** [review]
### Hi @melkamu12 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✔️ Good job implementing changes
✔️Linters are passing
✔️Very professional README
✔️Correct Git flow
✔️Descriptive PR
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**